src/ucs/async/thread.c

```diff
@@ -345,16 +347,12 @@ static ucs_status_t ucs_async_thread_modify_event_fd(ucs_async_context_t *async,
                                                      int event_fd, int events)
 {
     ucs_async_thread_t *thread = ucs_async_thread_global_context.thread;
-    struct epoll_event event;
     int ret;
 
-    memset(&event, 0, sizeof(event));
-    event.events  = events;
-    event.data.fd = event_fd;
-    ret = epoll_ctl(thread->epfd, EPOLL_CTL_MOD, event_fd, &event);
-    if (ret < 0) {
-        ucs_error("epoll_ctl(epfd=%d, ADD, fd=%d) failed: %m", thread->epfd,
-                  event_fd);
+    /* Store file descriptor into void * storage without memory allocation. */
+    ret = ucs_event_set_mod(thread->event_set, event_fd, events,
```

Comment:
need to convert events from POLLIN/POLLOUT/.. etc. to event_set constatns

Reply:
ucs_event_set_mod convert `UCS_EVENT_SET_READ` to `EPOLLIN` in that function.
So, I think it doesn't need to convert this.


```C
ucs_status_t ucs_event_set_mod(ucs_sys_event_set_t *event_set, int event_fd,
                               ucs_event_set_type_t events, void *callback_data)
{

    /* snip */
    
    raw_event.events   = ucs_event_set_map_to_raw_events(events);
```

@dmitrygladkov What do you think? 
I have one concern. How API user set `event` parameter in `ucs_async_ops_t.add_event_fd` function?

Reply:
@hiroyuki-sato let me investigate this issue
POLLOUT/POLLIN/POLLERR can be passed as events

Reply:
@yosefe @hiroyuki-sato 
we pass `POLL*` events to poll/epoll_ctl/kqueue (when FreeBSD and macOS support will be added)
it works with epoll, because `POLL*` constant have similiar values as `EPOLL*`
but I'm not sure whether this is a case for kqueue
I think we should fix it (not in this PR)

@hiroyuki-sato pls, handle `POLLERR` constant - add `UCS_EVENT_SET_ERROR`and map it to `EPOLLERR`
and we should convert from `POLL*` to `UCS_EVENT_SET_*` before passing to `ucs_event_set_mod` in `ucs_async_thread_modify_event_fd`
and `ucs_event_set_convert_poll_events` function or smth like that in UCS event set

Reply:
Hello, @dmitrygladkov 

Thank you for investigating this feature.
Do we need to do the following steps?
And Should I create new PR?

1. Add `UCS_EVENT_SET_ERROR` in `ucs_event_set_type_t`
2. Create new function `poll_events_convert_ucs_event_sets` in `ucs/async/thread.c`
   for converting `POLLIN` -> `UCS_EVENT_SET_READ`, `POLLOUT` -> `UCS_EVENT_SET_WRITE` otherwise `UCS_EVENT_SET_ERROR`
   Your proposal function name seems to me inverse.
3. Do we need to create inverse function? `EVENT_SET*` -> `POLL*`?
4. use 2. function  before call `ucs_event_set_mod` in ucs_async_thread_modify_event_fd
5. Use the same conversion 4 before `ucs_event_set_add` in `ucs_async_thread_add_event_fd`

Reply:
Hello, @dmitrygladkov. I pushed a new commit. PTAL